metricflow/dataflow/builder/dataflow_plan_builder.py

```diff
@@ -1935,3 +1997,9 @@ def _determine_time_spine_join_spec(
                 time_granularity=join_spec_grain, date_part=None
             )
         return join_on_time_dimension_spec
+
+    def _offset_window_is_custom(self, offset_window: Optional[MetricTimeWindow]) -> bool:
+        return (
+            offset_window is not None
+            and offset_window.granularity in self._semantic_model_lookup.custom_granularity_names
```

Comment:
What happens when the granularity is not in `custom_granularity_names`? Is this handled through a validation?